x-pack/plugins/security_solution/public/timelines/containers/helpers.ts

```diff
@@ -4,14 +4,32 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import isEqual from 'lodash/isEqual';
 import { TimelineId } from '../../../common/types/timeline';
 
 export const detectionsTimelineIds = [
   TimelineId.detectionsPage,
   TimelineId.detectionsRulesDetailsPage,
 ];
 
-export const skipQueryForDetectionsPage = (id: string, defaultIndex: string[]) =>
-  id != null &&
+/**
+ * If the defaultIndex for timeline and the defaultKibana index are equal to each other
+ * then this does not contain the signals index and we would want to exclude it completely from a query
+ * while on the detections page or the detection details page. This is to prevent additional querying
+ * from happening with the detections page and detections details when the default index is pushed into timeline.
+ * This code is considered a workaround and should go away in the future after refactoring/rewrites of timeline.
+ * @param id The id of the page which if it is set to the detections page or details page then it should not be analyzed
+ * @param defaultIndex The default index that is about cause a query from timeline.
+ * @param defaultKibanaIndex The default Kibana index to check to see if they're the same thing and if so, let us not query
+ */
+export const skipQueryForDetectionsPage = ({
+  id,
+  defaultIndex,
+  defaultKibanaIndex,
+}: {
+  id: string;
+  defaultIndex: string[];
+  defaultKibanaIndex: string[];
+}) =>
   detectionsTimelineIds.some((timelineId) => timelineId === id) &&
-  !defaultIndex.some((di) => di.toLowerCase().startsWith('.siem-signals'));
+  isEqual(defaultIndex.sort(), defaultKibanaIndex.sort());
```

Comment:
I do not think we should do it like that since we are going to bring kibana index patterns and we might have more index than the defaultKibanaIndex. and it will fail agains

Reply:
and these tests won't help to realize that we have a bug back

Reply:
I am thinking more about this one, I think we should use the value of `signalIndexName` from `useSignalIndex` and check that with `defaultIndex`. I feel like it might be more bulletproof that what we are offering/offered.

Reply:
I investigated that and since parts of the timeline are still using classes I will have to move upwards and use that hook before the class and then drill it down since hooks cannot be used directly in classes.

I was trying to avoid introducing more complexity and tech debt within timeline as this felt more like a temp workaround we're doing here already and the better fix is that timeline knows when it has its correct index being set before conducting its query generically.

I figured this is all just temporary code in other words and was trying to be careful to not add more complexity and debt by introducing more drill props into timeline within this PR as my primary mission.

With all that said, yes, if everyone feels comfortable with another drill prop and a this extra piece of data flowing downwards, I can change this with a pull request and add the additional drill props and the hook above the class.

Reply:
I will be fine with it, it is just a string.